lib/pbench/test/unit/server/conftest.py

```diff
@@ -120,12 +127,62 @@ def server_config(on_disk_server_config) -> PbenchServerConfig:
     return server_config
 
 
+@pytest.fixture(scope="session")
+def rsa_keys():
+    """Fixture for generating an RSA public / private key pair.
+
+    Returns:
+        A dictionary containing the RSAPrivateKey object, the PEM encoded public
+        key string without the BEGIN/END bookends (mimicing what is returned by
+        an OpenID Connect broker), and the PEM encoded public key string with
+        the BEGIN/END bookends
+    """
+    private_key = rsa.generate_private_key(public_exponent=65537, key_size=2048)
+    pem_public_key = (
+        private_key.public_key()
+        .public_bytes(Encoding.PEM, PublicFormat.SubjectPublicKeyInfo)
+        .decode()
+    )
```

Comment:
Yeah, you are right but I guess the test ensures that the code can correctly encrypt and decrypt messages using an RSA key-pair. While we could use a mock object to simulate the behavior of the key-pair, this would not guarantee that the code can work correctly with a real RSA key-pair. I am just moving this code that Peter added earlier, if we want to update this to a mock object I can open an issue work on it post this PR.